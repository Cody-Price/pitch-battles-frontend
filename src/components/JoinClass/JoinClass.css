.join-class-form {
  display: flex;
  width: 400px;
  height: 20px;
  margin: 0 2rem;
}

.join-class-form p {
  margin: 3px 0 0 0;
  font-size: 1.2rem;
}

.join-class-form input {
  font-family: "chronotype";
  font-size: 1.2rem;
}

.class-key-input {
  background: #e3e8d9;
  border: none;
  height: 18px;
  margin-right: 1rem;
  margin-left: 0.5rem;
  padding-left: 8px;
  box-shadow: 2px 2px 2px#4776bb;
}

.class-key-input:focus {
  outline: none;
}

.join-class-submit {
  font-family: "chronotype";
  font-size: 1.2rem;
  background: #4776bb;
  border: none;
  box-shadow: 2px 2px 2px#000;
  transition: 0.3s;
  padding-top: 4px;
}

.join-class-submit:active {
  box-shadow: none;
}
.join-class-submit:focus {
  outline: none;
}

@media (max-width: 860px) {
  .join-class-form {
    display: block;
    width: 180px;
    margin-left: 1rem;
  }

  .class-key-input,
  .join-class-submit {
    display: block;
  }

  .join-class-submit {
    margin: 0.5rem auto;
    width: 75px;
  }

  input.class-key-input {
    margin: 0;
  }

  .join-class-form p {
    margin: 0.5rem 0;
    text-align: center;
  }
}
