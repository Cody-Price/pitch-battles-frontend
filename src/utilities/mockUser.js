const mockUser = {
  first_name: "Chet",
  last_name: "Manly",
  avatar: 10,
  role: 0,
  email: "cmanly@gmail.com",
  games_played: 100,
  class: "6th-grade band"
};

export default mockUser;
