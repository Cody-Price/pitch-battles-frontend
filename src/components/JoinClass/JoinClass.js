import React, { Component } from "react";
import "./JoinClass.css";
import PropTypes from 'prop-types';

import { joinClassFetch } from "../../utilities/fetchCalls";

class JoinClass extends Component {
  constructor() {
    super();
    this.state = {
      classKey: ""
    };
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = async event => {
    event.preventDefault();

    try {
      const data = await joinClassFetch(
        this.state.classKey,
        this.props.user.id,
        this.props.webToken
      );

      this.props.getUpdatedUserData();
    } catch (error) {
      console.log(error);
    }
  };

  render() {
    return (
      <form onSubmit={e => this.handleSubmit(e)} className="join-class-form">
        <p>Join a class - </p>
        <input
          type="text"
          name="classKey"
          className="class-key-input"
          value={this.state.classKey}
          placeholder="class key"
          onChange={e => this.handleChange(e)}
        />
        <button className="join-class-submit">join</button>
      </form>
    );
  }
}

export default JoinClass;

JoinClass.propTypes = {
  id: PropTypes.number,
  webToken: PropTypes.string,
  getUpdatedUserData: PropTypes.func
}
