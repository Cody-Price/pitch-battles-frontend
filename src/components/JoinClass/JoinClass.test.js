import React from "react";
import mockUser from "../../utilities/mockUser";
import JoinClass from "./JoinClass";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("JoinClass", () => {
  let wrapper;
  let mockGetUpdatedUserData;
  const mockWebToken = "web toooken";

  beforeEach(() => {
    mockGetUpdatedUserData = jest.fn();
    wrapper = shallow(
      <JoinClass
        user={mockUser}
        webToken={mockWebToken}
        getUpdatedUserData={mockGetUpdatedUserData}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleChange", () => {
    it("should call setState", () => {
      const mockObject = {
        target: {
          name: "classKey",
          value: "fake value"
        }
      };
      wrapper.instance().handleChange(mockObject);

      expect(wrapper.state().classKey).toEqual("fake value");
    });

    it("should call handleChange on change", () => {
      const mockObject = {
        target: {
          name: "classKey",
          value: "fake value"
        }
      };
      const spy = jest.spyOn(wrapper.instance(), "handleChange");

      wrapper.instance().forceUpdate();

      wrapper.find(".class-key-input").simulate("change", mockObject);

      expect(spy).toHaveBeenCalledWith(mockObject);
    });
  });

  describe("handleSubmit", () => {
    it("should call handleSubmit on submit", async () => {
      let mockPreventDefault = jest.fn();
      const mockEvent = {
        preventDefault: mockPreventDefault
      };

      const spy = jest.spyOn(wrapper.instance(), "handleSubmit");

      wrapper.instance().forceUpdate();

      await wrapper.find(".join-class-form").simulate("submit", mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });

    it("should call preventDefault", async () => {
      let mockPreventDefault = jest.fn();
      const mockEvent = {
        preventDefault: mockPreventDefault
      };

      await wrapper.instance().handleSubmit(mockEvent);

      expect(mockPreventDefault).toHaveBeenCalled();
    });

    it("should call getUpdatedUserData", async () => {
      let mockPreventDefault = jest.fn();
      const mockEvent = {
        preventDefault: mockPreventDefault
      };

      await wrapper.instance().handleSubmit(mockEvent);

      expect(mockGetUpdatedUserData).toHaveBeenCalled();
    });
  });
});
