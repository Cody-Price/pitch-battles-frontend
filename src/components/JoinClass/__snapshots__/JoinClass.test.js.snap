// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinClass should match the snapshot 1`] = `
<form
  className="join-class-form"
  onSubmit={[Function]}
>
  <p>
    Join a class - 
  </p>
  <input
    className="class-key-input"
    name="classKey"
    onChange={[Function]}
    placeholder="class key"
    type="text"
    value=""
  />
  <button
    className="join-class-submit"
  >
    join
  </button>
</form>
`;
